#! /bin/sh /usr/share/dpatch/dpatch-run
## HostNameLookups_values.dpatch by Martin Pitt <mpitt@debian.org>
##
## DP: Fix possible values of HostNameLookups to comply to manpage and to be
## DP: consistent with other options. (Debian #379014, STR #2861)

@DPATCH@
diff -urNad trunk~/scheduler/conf.c trunk/scheduler/conf.c
--- trunk~/scheduler/conf.c	2008-06-12 17:32:30.000000000 +0000
+++ trunk/scheduler/conf.c	2008-06-12 17:34:06.000000000 +0000
@@ -2884,9 +2884,11 @@
       * Do hostname lookups?
       */
 
-      if (!strcasecmp(value, "off"))
+      if (!strcasecmp(value, "off") || !strcasecmp(value, "no") ||
+               !strcasecmp(value, "false"))
         HostNameLookups = 0;
-      else if (!strcasecmp(value, "on"))
+      else if (!strcasecmp(value, "on") || !strcasecmp(value, "yes") ||
+               !strcasecmp(value, "true"))
         HostNameLookups = 1;
       else if (!strcasecmp(value, "double"))
         HostNameLookups = 2;
